python/ray/serve/scripts.py

```diff
@@ -504,39 +505,92 @@ def shutdown(address: str, yes: bool):
         "If not provided, the config will be printed to STDOUT."
     ),
 )
+@click.option(
+    "--multi-app",
+    "-m",
+    is_flag=True,
+    help="Generate a multi-application config from multiple targets.",
+)
 def build(
-    import_path: str, app_dir: str, kubernetes_format: bool, output_path: Optional[str]
+    import_paths: Tuple[str],
+    app_dir: str,
+    kubernetes_format: bool,
+    output_path: Optional[str],
+    multi_app: bool,
 ):
     sys.path.insert(0, app_dir)
 
-    node: Union[ClassNode, FunctionNode] = import_attr(import_path)
-    if not isinstance(node, (ClassNode, FunctionNode)):
-        raise TypeError(
-            f"Expected '{import_path}' to be ClassNode or "
-            f"FunctionNode, but got {type(node)}."
-        )
+    def build_app_config(import_path: str, name: str = None):
+        node: Union[ClassNode, FunctionNode] = import_attr(import_path)
+        if not isinstance(node, (ClassNode, FunctionNode)):
+            raise TypeError(
+                f"Expected '{import_path}' to be ClassNode or "
+                f"FunctionNode, but got {type(node)}."
+            )
 
-    app = build_app(node)
-    schema = ServeApplicationSchema(
-        import_path=import_path,
-        runtime_env={},
-        host="0.0.0.0",
-        port=8000,
-        deployments=[deployment_to_schema(d) for d in app.deployments.values()],
-    )
+        app = build_app(node)
+        schema = ServeApplicationSchema(
+            import_path=import_path,
+            runtime_env={},
+            deployments=[
+                deployment_to_schema(d, not multi_app) for d in app.deployments.values()
+            ],
+        )
+        # If building a multi-app config, auto-generate names for each application.
+        # Also, each ServeApplicationSchema should not have host and port set, it should
+        # be set at the top level of ServeDeploySchema.
+        if multi_app:
+            schema.name = name
+            schema.route_prefix = app.ingress.route_prefix
```

Comment:
Can we verify somewhere in `serve build` that no two apps have the same `name` or `route_prefix`?

Reply:
Added parsing for validation!